test/e2e/common.go

```diff
@@ -338,24 +330,119 @@ func HasRootOnDisk(output string) bool {
 	return false
 }
 
-// IsBootedFromDisk checks if the system, accessed via the provided ssh.Client, is booted
-// from a disk. It executes the 'df -h' command on the remote system to analyze the filesystem
-// layout. In the case of a disk boot, the output includes a disk-based root filesystem
-// (e.g., '/dev/vda1'). Conversely, in the case of a Live-ISO boot, the primary filesystems
-// are memory-based (tmpfs).
-func IsBootedFromDisk(client *ssh.Client) (bool, error) {
+// CheckBootSource determines whether the system, accessed via the provided ssh.Client,
+// is booted from a disk or a live ISO. It executes the 'df -h' command on the remote system
+// to analyze the filesystem layout. A disk-based root filesystem (e.g., '/dev/vda1')
+// indicates a disk boot, while memory-based filesystems (tmpfs) indicate a Live-ISO boot.
+func CheckBootSource(client *ssh.Client) (bootedFromDisk bool, err error) {
 	cmd := "df -h"
 	output, err := executeSSHCommand(client, cmd)
 	if err != nil {
 		return false, fmt.Errorf("error executing 'df -h': %w", err)
 	}
 
-	isDisk := HasRootOnDisk(output)
-	if isDisk {
+	bootedFromDisk = HasRootOnDisk(output)
+	if bootedFromDisk {
 		capm3_e2e.Logf("System is booted from a disk.")
 	} else {
 		capm3_e2e.Logf("System is booted from a live ISO.")
 	}
 
-	return isDisk, nil
+	return bootedFromDisk, nil
+}
+
+func EstablishSSHConnection(authMethod, user, address string, useKeyAuth bool) *ssh.Client {
```

Comment:
How about we make the `authMethod` an actual `ssh.AuthMethod`? Then you can skip the `useKeyAuth` parameter and this becomes much cleaner.